addons/web/static/src/webclient/switch_company_menu/switch_company_menu.js

```diff
@@ -2,65 +2,58 @@
 
 import { Dropdown } from "@web/core/dropdown/dropdown";
 import { DropdownItem } from "@web/core/dropdown/dropdown_item";
-import { useService } from "@web/core/utils/hooks";
 import { registry } from "@web/core/registry";
-import { browser } from "@web/core/browser/browser";
-
-import { Component, useState, reactive } from "@odoo/owl";
-
-const store = reactive({
-    toggleTimer: null,
-    nextAvailableCompanies: [],
-});
 
+import { Component } from "@odoo/owl";
+import { debounce } from "@web/core/utils/timing";
+import { useCompanyState } from "@web/webclient/company_service";
 
 export class SwitchCompanyItem extends Component {
+    static template = "web.SwitchCompanyItem";
+    static components = { DropdownItem, SwitchCompanyItem };
+    static props = {
+        company: {},
+        level: { type: Number },
+    };
+    static toggleDelay = 1000;
+
     setup() {
-        this.companyService = useService("company");
-        this.store = useState(store);
+        this.companyService = useCompanyState();
+        this.debouncedApply = debounce(
```

Comment:
this debounce logic of committing the state could be moved to the service as well

Reply:
Actually, I think it would give more sense to the selectedCompanies attribute, as the debounce is (as far as I understand) the reason why we need it